### PR TITLE
Detaches Anonymous themes and Triple AI from SSTicker (+ more anonymous themes and content)

### PR DESCRIPTION
**Pull request by tralezab**
## About The Pull Request

Moved both of those features off SSTicker, though they still use it to check pregame which is unavoidable

Ai triumvirate is now a singleton datum controller, anonymous theme datums now handle ALL the logic on themselves instead of some of it being on the admin button and global procs

Also, I dunno, while I was working on anonymous names I added some content. Some of the modes now have optional extras you can throw in (stuff that i've had to adminbus to add every time) and there's two more anonymous themes.

## Why It's Good For The Game

SSTicker is a deprecated subsystem that @Mothblocks is trying to remove at some point in the future. Thought I'd correct my mistake of putting anonymous names there and also move tripAI off as well

## Changelog
:cl:
add: anonymous names feature content
spellcheck: any reference to ai triumvirate as "tripAI" has been obliterated. use the right case next time, jagoff
code: Moved ai triumvirate off SSTicker
refactor: ai triumvirate is now a singleton datum controller
admin: anonymous names has two new themes and some optional enabled theme features
/:cl:

### CODE REVIEW

File: code/modules/admin/verbs/ai_triumvirate.dm [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+		to_chat(usr, "This option is currently only usable during pregame. This may change at a later date.", confidential = TRUE)
+		return
+
+	var/datum/job/job = SSjob.GetJob("AI")
+	if(!job)
+		to_chat(usr, "Unable to locate the AI job", confidential = TRUE)
+		return
+
+	GLOB.triple_ai_controller = new()
+	to_chat(usr, "There will [GLOB.triple_ai_controller ? "" : "not"] be an AI Triumvirate at round start.")
```
Comment (coiax) [inline code/modules/admin/verbs/ai_triumvirate.dm:44]: The wording implies that this is a toggle, but this is more like an Enable with no option to turn it off.

Comment (tralezab) [reply code/modules/admin/verbs/ai_triumvirate.dm:44]: It should be a toggle actually, my bad.

File: code/modules/admin/verbs/ai_triumvirate.dm [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+/datum/triple_ai_controller
+
+/datum/triple_ai_controller/New()
+	. = ..()
+	RegisterSignal(SSjob, COMSIG_OCCUPATIONS_DIVIDED, .proc/on_occupations_divided)
+
+/datum/triple_ai_controller/proc/on_occupations_divided(datum/source)
+	SIGNAL_HANDLER
+
+	for(var/datum/job/ai/ai_datum in SSJob.occupations)
```
Comment (Mothblocks) [inline code/modules/admin/verbs/ai_triumvirate.dm:19]: You made sure to test this, right?

```suggestion
	for(var/datum/job/ai/ai_datum in SSjob.occupations)
```

File: code/modules/admin/verbs/ai_triumvirate.dm (hunk 1 above)
Comment (Mothblocks) [inline code/modules/admin/verbs/ai_triumvirate.dm:44]:
```suggestion
	to_chat(usr, "There will[GLOB.triple_ai_controller ? "" : "not"] be an AI Triumvirate at round start.")
```

Comment (Mothblocks) [reply code/modules/admin/verbs/ai_triumvirate.dm:44]: Otherwise, two spaces.

File: code/modules/admin/verbs/anonymousnames.dm [hunk 3]
```diff
@@ -35,24 +25,68 @@
 	var/result = input(usr, "Choose an anonymous theme","going dark") as null|anything in input_list
 	if(!usr || !result || result == "Cancel")
 		return
-	result = input_list[result]
+	var/datum/anonymous_theme/chosen_theme = input_list[result]
+	var/extras_enabled = "No"
 	var/alert_players = "No"
 	if(SSticker.current_state > GAME_STATE_PREGAME) //before anonnames is done, for asking a sleep
+		if(initial(chosen_theme.extras_enabled))
+			extras_enabled = tgui_alert(usr, extras_enabled, "2016 admins didn't miss roundstart", list("Yes", "No"))
```
Comment (Mothblocks) [inline code/modules/admin/verbs/anonymousnames.dm:33]: ?

Comment (Mothblocks) [reply code/modules/admin/verbs/anonymousnames.dm:33]: I have literally no idea what this means

File: code/modules/admin/verbs/anonymousnames.dm [hunk 4]
```diff
@@ -62,30 +96,49 @@
...
+ */
+/datum/anonymous_theme/proc/restore_all_players()
+	priority_announce("Names and Identities have been restored.", "Identity Restoration", SSstation.announcer.get_rand_alert_sound())
+	for(var/mob/living/player in GLOB.player_list)
+		if(!player.mind || (!ishuman(player) && !issilicon(player)) || !SSjob.GetJob(player.mind.assigned_role))
+			continue
+		var/old_name = player.real_name //before restoration
+		if(issilicon(player))
+			var/is_AI = isAI(player)
+			player.apply_pref_name("[is_AI ? "ai" : "cyborg"]", player.client)
```
Comment (Mothblocks) [inline code/modules/admin/verbs/anonymousnames.dm:115]:
```suggestion
			player.apply_pref_name("[isAI(player) ? "ai" : "cyborg"]", player.client)
```

**Review by Mothblocks — CHANGES_REQUESTED** [review]

File: code/modules/admin/verbs/ai_triumvirate.dm (hunk 2 above)
Comment (tralezab) [reply code/modules/admin/verbs/ai_triumvirate.dm:19]: I fixed this but forgot to push it

File: code/modules/admin/verbs/anonymousnames.dm (hunk 3 above)
Comment (tralezab) [reply code/modules/admin/verbs/anonymousnames.dm:33]: hmm... something needs to be rewritten here

**Pull request merged**